### PR TITLE
Fix pattern expression with same edge variable

### PR DESCRIPTION
**Pull request by czpmango**
## What type of PR is this?
- [x] bug
- [ ] feature
- [ ] enhancement

#### Issue(s) number: 
https://github.com/vesoft-inc/nebula-ent/issues/2060

## Checklist:
Tests:
- [ ] Unit test(positive and negative cases)
- [ ] Function test
- [ ] Performance test
- [ ] N/A

Affects:
- [ ] Documentation affected (Please add the label if documentation needs to be modified.)
- [ ] Incompatibility (If it breaks the compatibility, please describe it and add the label.）
- [ ] If it's needed to cherry-pick (If cherry-pick to some branches is required, please label the destination version(s).)
- [ ] Performance impacted: Consumes more CPU/Memory

## Release notes:

Please confirm whether to be reflected in release notes and how to describe:
> ex. Fixed the bug .....

### CODE REVIEW

**Comment by Shylock-Hg** [comment]
Add more test in `feature/match/PathExprRefLocalVariable.feature`

**Comment by czpmango** [comment]
> Add more test in `feature/match/PathExprRefLocalVariable.feature`

OKey.

File: tests/tck/features/match/PathExpr.feature [hunk 1]
```diff
@@ -50,22 +50,22 @@ Feature: Basic match
       """
       MATCH (v:player) WITH (v)-[v]-() AS p RETURN p
       """
-    Then a SemanticError should be raised at runtime: Alias `v' should be Edge, but got type 'Node'
```
Comment (Shylock-Hg) [inline tests/tck/features/match/PathExpr.feature:53]: Why change this?

Comment (czpmango) [reply tests/tck/features/match/PathExpr.feature:53]: The previous error message is a bit ambiguous.

File: tests/tck/features/match/PathExprRefLocalVariable.feature [hunk 2]
```diff
@@ -100,6 +100,27 @@ Feature: Path expression reference local defined variables
       | name         |
       | "Tim Duncan" |
       | "Tim Duncan" |
+    When executing query:
```
Comment (Shylock-Hg) [inline tests/tck/features/match/PathExprRefLocalVariable.feature:103]: Should add test in different clauses.

**Review by xtcyclist — APPROVED** [review]

File: tests/tck/features/match/PathExprRefLocalVariable.feature (hunk 2 above)
Comment (czpmango) [reply tests/tck/features/match/PathExprRefLocalVariable.feature:103]: ACK.

**Review by Shylock-Hg — APPROVED** [review]

**Review by xtcyclist — APPROVED** [review]

**Pull request merged**